python/pyarrow/table.pxi

```diff
@@ -3436,8 +3449,18 @@ cdef class RecordBatch(_Tabular):
                                  '{0} vs {1}'.format(len(arr), num_rows))
             c_arrays.push_back(arr.sp_array)
 
-        result = pyarrow_wrap_batch(CRecordBatch.Make(c_schema, num_rows,
-                                                      c_arrays))
+        if c_arrays.size() > 0:
+            c_device_type = deref(c_arrays[0]).device_type()
+            result = pyarrow_wrap_batch(CRecordBatch.MakeWithDevice(c_schema,
+                                                                    num_rows,
+                                                                    c_arrays,
+                                                                    c_device_type,
```

Comment:
We are assuming here that then all arrays have the same device. I think that is the same assumption as in C++? (where this is only verified in debug mode?) In which case that is probably fine for now?

Reply:
Correct, I was following C++'s design choice. However, this can now be removed!